clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/ModifyAsgAtomicOperation.groovy

```diff
@@ -78,6 +80,20 @@ class ModifyAsgAtomicOperation implements AtomicOperation<Void> {
           .withHealthCheckType(description.healthCheckType)
           .withTerminationPolicies(description.terminationPolicies)
 
+      if (description.disabledGroupMetrics) {
```

Comment:
would it be possible to just make the API payload contain the list of desired enabled metrics, and do a describe call to see what if anything is currently enabled and the figure out the appropriate set of disable and enable calls to issue?

It would be nice to keep the API payload as just making a request saying 'this is what I want the thing to look like, make it so'

We do something similar in ModifyLaunchConfiguration - if the request has instance metrics collection enabled, we figure out the current state and adjust the calls accordingly to achieve the requested state